1. Implemented movie certification and TV show content ratings.

wagnered opened the pull request:
2. Implemented Movie and TV show overviews and TV episode summary
3. Added ability to search db for movies and TV shows.
4. Added improved filename parsing of filenames following partial kodi (xmbc)
   formats.
5. Eliminated redundant gather_art calls to Tmdb/Tvdb plugins.
4. Changed config version to 30 in config file and init.php.
5. Changed database version to 380000 because of additions/modifications to tables.
6. Changed height to auto in default.css for dl.medial.details dd to accommodate
   varying height of overview and summaries.

paulblei commented on lib/class/movie.class.php:
```diff
@@ -64,19 +64,14 @@ public static function gc()
      * This takes a key'd array of data as input and inserts a new movie entry, it returns the record id
      */
-    public static function insert(array $data, $gtypes = array(), $options = array())
+   public static function insert(array $data, $gtypes = array(), $options = array())
```
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Avoid unused parameters such as '$gtypes'.](https://www.codacy.com/app/paul/ampache/file/616134461/issues/source?bid=7631&fileBranchId=2290754#l67)

paulblei commented on lib/class/tvshow.class.php:
```diff
@@ -359,12 +354,8 @@ public function update(array $data)
+        Dba::write($sql, array($name, $year, $content_rating, $overview, $current_id));
 
         $this->name = $name;
         $this->prefix = $prefix;
```
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Avoid unused local variables such as '$prefix'.](https://www.codacy.com/app/paul/ampache/file/616134494/issues/source?bid=7631&fileBranchId=2290754#l361)

paulblei commented on lib/class/search.class.php:
```diff
@@ -1386,6 +1434,157 @@ private function video_to_sql()
+     *
+     * Handles the generation of the SQL for TV show searches.
+     */
+    private function tvshow_to_sql()
```
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method tvshow_to_sql is not named in camelCase.](https://www.codacy.com/app/paul/ampache/file/616134479/issues/source?bid=7631&fileBranchId=2290754#l1521)

paulblei commented on lib/class/update.class.php:
```diff
@@ -3362,126 +3355,41 @@ public static function update_370030()
-     * Add an option to customize login art, favicon and text footer.
      */
-    public static function update_370031()
+    public static function update_380000()
```
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method update_380000 is not named in camelCase.](https://www.codacy.com/app/paul/ampache/file/616134498/issues/source?bid=7631&fileBranchId=2290754#l3366)

paulblei replied:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method update_380000() has an NPath complexity of 78125. The configured NPath complexity threshold is 200.](https://www.codacy.com/app/paul/ampache/file/616134498/issues/source?bid=7631&fileBranchId=2290754#l3366)

paulblei commented on lib/class/search.class.php:
```diff
@@ -1386,6 +1434,157 @@ private function video_to_sql()
+     *
+     * Handles the generation of the SQL for video searches.
+     */
+    private function movie_to_sql()
```
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method movie_to_sql() has a Cyclomatic Complexity of 11. The configured cyclomatic complexity threshold is 10.](https://www.codacy.com/app/paul/ampache/file/616134479/issues/source?bid=7631&fileBranchId=2290754#l1441)

paulblei replied:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Avoid unused private methods such as 'movie_to_sql'.](https://www.codacy.com/app/paul/ampache/file/616134479/issues/source?bid=7631&fileBranchId=2290754#l1441)

paulblei commented on lib/class/tvshow_episode.class.php:
```diff
@@ -127,8 +127,8 @@ public function update(array $data)
         $summary = isset($data['summary']) ? $data['summary'] : $this->summary;
-
-        $sql = "UPDATE `tvshow_episode` SET `original_name` = ?, `season` = ?, `episode_number` = ?, `summary` = ? WHERE `id` = ?";
+        $content_rating = isset($data['content_rating']) ? $data['content_rating'] : $this->content_rating;
```
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Avoid unused local variables such as '$content_rating'.](https://www.codacy.com/app/paul/ampache/file/616134495/issues/source?bid=7631&fileBranchId=2290754#l130)

paulblei commented on lib/class/movie.class.php:
```diff
@@ -86,19 +81,12 @@ public function update(array $data)
+        Dba::write($sql, array($data['original_name'], $summary, $year, $certification, $this->id));
 
         $this->original_name = $name;
         $this->prefix = $prefix;
```
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Avoid unused local variables such as '$prefix'.](https://www.codacy.com/app/paul/ampache/file/616134461/issues/source?bid=7631&fileBranchId=2290754#l92)

wagnered commented:
I see codacy  has it's drawbacks:  It doesn't consider user function calls.
I removed the unused $prefix and $gtypes parameter. and will wait to hear your comment on the following before I commit the changes.

Other than possibly changing function update_380000() to camelcase in update.class.php, I don't think movie_to_sql and tvshow_to_sql should be camel cased, because of all the other dependent changes that would be required.

Afterster commented:
Indeed Codacy is not perfect in this case and I agree with you about the camelcase non-sense here.
I'm on the road for few days, I will review your code next week I think. Thanks!

paulblei commented:
You're right, and I'm sorry for these comments. I wrongly activated those on my Codacy project. 
I deleted them now

Afterster commented:
Ok I got some time to review your PR.
 - Settings specific to a plugin shouldn't be global (tmdb_secure)
 - Why removing get_description, display_art and remove_from_disk functions in tvshow_episode.class.php ?
 - Same for other changes made recently in the develop branch (update.class.php, ...), I believe you failed the merge.

wagnered commented:
Hi @Afterster,

First Question:  Where should i put it so that it isn't global?  It
seems to me that the only other place is Server Preferences/system.
Should tmdb_secure (https) be available to content/catalog managers?

As to the second question, I actually don't remember removing those
functions.  It must have been early in the modifications.

1.  I couldn't seem to find any reference to "remove_from_disk" anywhere
else in the code. (No calling functions) Second, the function references
parent::remove_from_disk which removes all episodes, seasons and video
info.  That doesn't seem correct for just wanting to remove one episode
if the calling logic was implemented.
	As I was recently thinking about implementing the removal of individual
episodes, seasons and movies or at least disable them like what was done
with songs, I will add the remove-from-disk function again,

2. As for "get_description" in tvshow_episode class, it is public, but
there are no functions, private or public calling this function. I don't
know how this function was intended to be used, but the tvshow overview
and tvshow_episode summary are displayed when the appropriate pages are
browsed.  I'll re-include it for possible future use, though.

3. Again, I'm not quite sure why I removed tvshow_episode::display_art,
possibly because there are only two types of art, default and preview,
it would not get stored anyway with the current logic. Only artwork for
movies and tvshows are capable of being stored with the current logic.
So this function seems to never be called.
	I guess my principles are to implement feature with the minimal amount
of modifications, but I'll put it back in for possible future use.

4. Because there are changes to the database and ampache.cfg.php.dist, I
assumed that it would call for updating the config file to 30 and the
database to 380000 to conform to the Amapche version number.

@Afterster, you wrote: "I believe you failed the merge." Could you give
me details as to how and what I need to do to fix things other than what
I wrote above,  if those corrections apply?


-- 
73 From Ernie D

AA1AD | CN87ug 

-----Original Message-----From: Afterster <notifications@github.com>
Reply-to: ampache/ampache <reply
+005d506e68cabe0bbed7128e008cf396ebdfd446e2d2a82592cf000000011169a01e92a169ce04653ab3@reply.github.com>
To: ampache/ampache <ampache@noreply.github.com>
Cc: Ernest Wagner <wagnered@comcast.net>
Subject: Re: [ampache] 1. Implemented movie certification and TV show
content ratings. (#846)
Date: Tue, 12 May 2015 04:03:58 -0700



Ok I got some time to review your PR.

      * Settings specific to a plugin shouldn't be global (tmdb_secure)
      * Why removing get_description, display_art and remove_from_disk
        functions in tvshow_episode.class.php ?
      * Same for other changes made recently in the develop branch
        (update.class.php, ...), I believe you failed the merge.

—
Reply to this email directly or view it on GitHub.

wagnered commented:
By "failing the merge" are you referring to the scrutinizer failure tests?  The build passed.  The hint indicates that every team member must have an account on scrutinizer.

Afterster commented:
About 1) 2) 3), you're right it is not really used (yet) but those are mandatory functions because of inheritance from `library_item` interface.
4) Yes you're right as soon the 3.8 got release we should bump to 380000 (one day... but as soon I fix a major issue I believe we should wait few more weeks, and unfortunately I'm still fixing major issues...).

Nop I'm not refering to scrutinizer with "failing the merge". I think you merged wrongly recent changes from base (I can't see it in your git history so I'm not totally sure). But the result looks wrong. Did you used rebase?
If you take a look at  lib/class/update.class.php file in the diff https://github.com/ampache/ampache/pull/846/files#diff-f7e4524758c7313f2163f30a3ef2d075, you will see for instance that your commit removed most of recent changes in develop branch. This is not excepted I believe.

wagnered commented:
I see what you mean, and all I got to say is "OMG", but I don't
understand how it happened. I did not do a rebase because I don't
understand the concept, but as I said earlier, I blew away the project
folder and had to start over. I started from a new fork of
ampache/ampache. 

The only commit/push I made was when I made the final changes and had
them fully functioning and ready for the PR.

What I did do was try to update my fork from ampache/ampache using
command line git, maybe that is where I want wrong.

So, I'll close my PR and start over.  I have no problem with that as I
think the features are well worth the effort.

-- 
73 From Ernie D

AA1AD | CN87ug 

-----Original Message-----From: Afterster <notifications@github.com>
Reply-to: ampache/ampache <reply
+005d506eb7f6f84c137134b4357df0e3cacb22f06e99ad6592cf00000001116e25e392a169ce04653ab3@reply.github.com>
To: ampache/ampache <ampache@noreply.github.com>
Cc: Ernest Wagner <wagnered@comcast.net>
Subject: Re: [ampache] 1. Implemented movie certification and TV show
content ratings. (#846)
Date: Fri, 15 May 2015 14:23:47 -0700



About 1) 2) 3), you're right it is not really used (yet) but those are
mandatory functions because of inheritance from library_item interface.
4) Yes you're right as soon the 3.8 got release we should bump to 380000
(one day... but as soon I fix a major issue I believe we should wait few
more weeks, and unfortunately I'm still fixing major issues...).

Nop I'm not refering to scrutinizer with "failing the merge". I think
you merged wrongly recent changes from base (I can't see it in your git
history so I'm not totally sure). But the result looks wrong. Did you
used rebase?
If you take a look at lib/class/update.class.php file in the diff
https://github.com/ampache/ampache/pull/846/files#diff-f7e4524758c7313f2163f30a3ef2d075, you will see for instance that your commit removed most of recent changes in develop branch. This is not excepted I believe.

—
Reply to this email directly or view it on GitHub.

wagnered commented:
Closing and starting over. ; ' >

wagnered commented:
@afterster,

A question and a comment:

When was prettifyTitle(urldecode($this->_pathinfo['filename'])) removed from Vainfo?  Because 
the statement *"$parsed['title'] = urldecode($this->_pathinfo['filename']);"* in $parse_general stores the entire filename, not the title.  what is parsed in parse_general supersedes the title that is downloaded from Tmdb. If we are going to conform to kodi (xbmc) and most other file naming standards, I would like to add back PrettifyTitle()  to extract the  title from the filename.  Either that, or rearrange the order that the tags are parsed in function clean_tag_info().

Is there a way to find out from the log when the prettifyTitle function was removed?

Also, this change made me realize what might be the cause of the mess up of this PR.  I had been storing my changes outside the repository and copying the whole file back into the repository, instead of cut/paste the changes after updating the fork/repository from remote.  I'm not sure how to handle this if there are a lot of changes going on from you or others with files i am also working on.

Afterster commented:
Mhh are you sure prettifyTitle ever existed in base? Looks like one of your new function.
About the merge issue, my advice for later would be to always use git function to do the merge, resolve conflict, or even switching between branches (instead of cut/copy/paste).

wagnered commented:
OK, I guess I'm getting old and forgetful.  I found a copy of that function in a backup.  I decided i didn't need it after all as for TV shows, I will be using 'episode_title' to display as title on the tv shows page.

"About the merge".  I don't know how to do any of those things without using  git functions.  ;' >
What I meant by copy/paste was that when I created a new github fork/local clone (starting over), I think I should have used copy/paste to make the specific changes and not just replace the newly cloned file as I had done. The changes have got to get into the clean file somehow, just not by replacing the whole file. ; ' >

In a way, I'm happy that I will be creating a new PR as removing 'original_name' in 'tvshow_episode' caused more required changes than I realized earlier.

Afterster commented:
Haha ok. Maybe creating git patches could be a solution here instead of copy/past?

wagnered commented:
@Afterster ,

After reworking this PR, I realized that the changes I made to the Tmdb module was the ones I submitted to the developer of the Tmdb module.  He immediately merged the PR into a different branch with several additional dependencies. In other words, the changes I made were for a dead-end branch, which ampache uses.

I was wondering if I should submit the PR with the original changes or the PR with the latest Tmdb version(2.0.7) and the supporting modules (GuzzleHttp/, Psr, React/Promise,  /Symfony/Component/OptionsResolver/,)

Afterster commented:
If we're using a dead branch then we should move for sanity reason yes...
This kind of dependency hell is really anoying, we should really handle #191 but it make it hard because of several dependencies without Composer package.

Anyway I took a look on the new dependencies, looks fine for me. Thanks.
